pydash/arrays.py

```diff
@@ -1322,6 +1348,31 @@ def sorted_index(array, value, callback=None):
     return bisect_left(array, value)
 
 
+def sorted_index_of(array, value):
+    """Returns the index of the matched `value` from the sorted `array`, else
+    -1.
+
+    Args:
+        array (list): Array to inspect.
+        value (mixed): Value to search for.
+
+    Returns:
+        int: Returns the index of the first matched value, else -1.
+
+    Example:
+
+        >>> sorted_index_of([10, 5, 7, 3], 3)
+        0
+        >>> sorted_index_of([10, 10, 5, 7, 3], 10)
+        3
+        >>> sorted_index_of([10, 5, 7, 3], 11)
+        -1
+
+    .. versionadded:: TODO
+    """
+    return sorted(array).index(value) if value in array else -1
```

Comment:
The documenatation in lodash is quite confusing to me. It says `performs a binary search on a sorted array`. So is the array already sorted or being sorted as part of the functionality?

I can see that the functionality is around an array which is already sorted. The array is not sorted as part of the functionality.